src/compile/mark/rule.ts

```diff
@@ -15,11 +16,23 @@ export const rule: MarkCompiler = {
       return {};
     }
 
+    let secondary = {};
```

Comment:
I added a cleaner fix for this in  https://github.com/vega/vega-lite/pull/3265.  
You can help review and merge the PR so you can rebase this branch.

Reply:
I think I still need this actually. Can you verify?

Reply:
I don't think it is needed. I removed your changes and the rule example still works.